stl/inc/chrono

```diff
@@ -5177,184 +5217,1318 @@ namespace chrono {
+        return _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:%F %T}"), _Val);
+    }
+
+    template <class _CharT, class _Traits, class _Duration>
+    basic_ostream<_CharT, _Traits>& operator<<(basic_ostream<_CharT, _Traits>& _Os, const local_time<_Duration>& _Val) {
+        return _Os << sys_time<_Duration>{_Val.time_since_epoch()};
+    }
+
+    template <class _CharT, class _Traits, class _Duration>
+    basic_ostream<_CharT, _Traits>& operator<<(
+        basic_ostream<_CharT, _Traits>& _Os, const _Local_time_format_t<_Duration>& _Val) {
+        // Doesn't appear in the Standard, but allowed by N4885 [global.functions]/2.
+        // Implements N4885 [time.zone.zonedtime.nonmembers]/2 for zoned_time.
+        return _Os << _STD format(_Os.getloc(), _STATICALLY_WIDEN(_CharT, "{:%F %T %Z}"), _Val);
+    }
+
+    template <class _CharT, class _Traits, class _Duration, class _TimeZonePtr>
+    basic_ostream<_CharT, _Traits>& operator<<(
+        basic_ostream<_CharT, _Traits>& _Os, const zoned_time<_Duration, _TimeZonePtr>& _Val) {
+        const auto _Info = _Val.get_info();
+        return _Os << _Local_time_format_t<_Duration>{_Val.get_local_time(), &_Info.abbrev};
+    }
+
+    template <class _CharT>
+    struct _Chrono_formatter {
+        _Chrono_formatter() = default;
+
+        explicit _Chrono_formatter(const basic_string_view<_CharT> _Time_zone_abbreviation_)
+            : _Time_zone_abbreviation{_Time_zone_abbreviation_} {}
+
+        template <class _Ty>
+        _NODISCARD auto _Parse(basic_format_parse_context<_CharT>& _Parse_ctx) {
+            _Chrono_specs_setter<_CharT, basic_format_parse_context<_CharT>> _Callback{_Specs, _Parse_ctx};
+            const auto _It =
+                _Parse_chrono_format_specs(_Parse_ctx._Unchecked_begin(), _Parse_ctx._Unchecked_end(), _Callback);
+            const auto _Res_iter = _Parse_ctx.begin() + (_It - _Parse_ctx._Unchecked_begin());
+
+            if (_It != _Parse_ctx._Unchecked_end() && *_It != '}') {
+                _THROW(format_error("Missing '}' in format string."));
+            }
+
+            if constexpr (_Is_specialization_v<_Ty, duration>) {
+                if constexpr (!treat_as_floating_point_v<typename _Ty::rep>) {
+                    if (_Specs._Precision != -1) {
+                        _THROW(format_error("Precision specification invalid for chrono::duration type with "
+                                            "integral representation type, see N4885 [time.format]/1."));
+                    }
+                }
+            } else {
+                if (_Specs._Precision != -1) {
+                    _THROW(format_error("Precision specification invalid for non-chrono::duration type, "
+                                        "see N4885 [time.format]/1."));
+                }
+            }
+
+            const auto& _List = _Specs._Chrono_specs_list;
+
+            // [time.format]/6

```

Comment:
This should cite the Working Paper (since it's referring to a specific paragraph and not an entire section), but I don't want to reset testing over this; will clean it up later if no further commits are necessary.